cvmfs/network/download.cc

```diff
@@ -2767,6 +2807,22 @@ void DownloadManager::UpdateProxiesUnlocked(const string &reason) {
   unsigned num_alive = (group->size() - opt_proxy_groups_current_burned_);
   string old_proxy = JoinStrings(opt_proxy_urls_, "|");
 
+  // only for non-sharding request get the host name of the old proxy
+  std::string old_host_name = "";
+  if (!opt_proxy_shard_ && opt_proxy_urls_.size() > 0
+      && opt_proxy_groups_->size() > 0) {
+      int prev_idx = opt_proxy_groups_current_ == 0 ?
+                        opt_proxy_groups_->size() - 1
+                        : opt_proxy_groups_current_ - 1;
+      vector<ProxyInfo> *prev_group = &((*opt_proxy_groups_)[prev_idx]);
+      for (size_t i = 0; i < prev_group->size(); i++) {
+        ProxyInfo *proxy = &(*prev_group)[i];
+        if (opt_proxy_urls_[0] == proxy->url) {
+          old_host_name = proxy->host.name();
+        }
+      }
+  }
```

Comment:
That one and the following block to get the host name of the new proxy look brittle to me. Any chance we can leave that out of the patch release?

Reply:
we can remove it and have a more detailed check for the 2.12 release
@vvolkl let me know how you want to do it (or if it is easier feel free to do it yourself)